public/app/core/components/TimeRangeShift/TimeRangeShift.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useState } from 'react';
+import { useDebounce } from 'react-use';
+
+import { InlineField, InlineFieldRow, Input } from '@grafana/ui';
+
+interface Props {
+  label: string;
+  tooltip: string;
+  value: string;
+  onChange: (val: string) => void;
+  isInvalid: boolean;
+  disabled?: boolean;
+}
+
+export function TimeRangeShift(props: Props) {
+  const [localValue, setLocalValue] = useState(props.value);
+
+  useDebounce(
```

Comment:
Setting the default value to whatever was previously saved, then only validating every 500ms so that the user doesn't get an error message immediately after typing any char and before they they are finished e.g. if they types 1 it would otherwise show an error before they are finished typing. If made changes but if there is an easier way to do this lmk